ggml/src/ggml-sycl/set_rows.cpp

```diff
@@ -0,0 +1,126 @@
+#include "set_rows.hpp"
+
+template<typename TIn, typename TOut>
+static inline void convert(const char* src, char* dst) {
+    auto src_val = *reinterpret_cast<const TIn*>(src);
+    auto dst_val = sycl::vec<TIn, 1>(src_val).template convert<TOut>()[0];
+    *reinterpret_cast<TOut*>(dst) = dst_val;
+}
```

Comment:
I think we should be using SFINAE here to not allow types we do not support (the enable_if_t part that was present in the snippet)

Reply:
```
error: no matching function for call to 'convert'
   50 |         convert<TIn, TOut>(src_elem, dst_elem);
note: in instantiation of function template specialization 'k_set_rows<float, sycl::detail::half_impl::half>' requested here
   78 |                 k_set_rows<TIn, TOut>(
      |                 ^
note: in instantiation of function template specialization 'set_rows_sycl<float, sycl::detail::half_impl::half>' requested here
  121 |             set_rows_sycl<float, sycl::half>(
      |             ^
note: candidate template ignored: requirement 'utils::is_arithmetic_v()' was not satisfied [with TIn = float, TOut = sycl::detail:
:half_impl::half]
   14 | convert(const char* src, char* dst) {
      | ^
```

Reply:
if you tested the suggested code above, there are a few small typos in there. Can you try the following? To me it compiles without any issue with icpx 2025.1 and icpx 2025.2 
```suggestion
namespace utils {
template<typename T>
constexpr bool is_arithmetic_v() {
    return std::is_arithmetic_v<T> || std::is_same_v<T, sycl::half> || std::is_same_v<T, sycl::ext::oneapi::bfloat16>;
}
}

template<typename TIn, typename TOut>
std::enable_if_t<utils::is_arithmetic_v<TIn>() && utils::is_arithmetic_v<TOut>(), void>
convert (const char* src, char* dst) {
    auto src_val = *reinterpret_cast<const TIn*>(src);
    // compiler should be able to convert this to a no-op where T1 = T2
    auto dst_val = sycl::vec<TIn, 1>(src_val).template convert<TOut, sycl::rounding_mode::automatic>()[0];
   *reinterpret_cast<TOut*>(dst) = dst_val;;
}
```